Client/Frontend/Browser/BraveSkusScriptHandler.swift

```diff
@@ -0,0 +1,144 @@
+
+import Foundation
+import WebKit
+import Shared
+import BraveShared
+import BraveCore
+import BraveVPN
+
+private let log = Logger.browserLogger
+
+class BraveSkusScriptHandler: TabContentScript {
+  typealias ReplyHandler = (Any?, String?) -> Void
+  
+  private weak var tab: Tab?
+  
+  private let sku: SkusSkusService?
+  
+  required init(tab: Tab) {
+    self.tab = tab
+    self.sku = Skus.SkusServiceFactory.get(privateMode: tab.isPrivate)
+  }
+  
+  static func name() -> String { "BraveSkusHelper" }
+  
+  func scriptMessageHandlerName() -> String? { "braveSkusHelper" }
+  
+  private enum Method: Int {
+    case refreshOrder = 1
+    case fetchOrderCredentials = 2
+    case prepareCredentialsPresentation = 3
+    case credentialsSummary = 4
+  }
+  
+  func userContentController(_ userContentController: WKUserContentController, didReceiveScriptMessage message: WKScriptMessage, replyHandler: @escaping (Any?, String?) -> Void) {
+    let allowedHosts = DomainUserScript.braveSkus.associatedDomains
+    
+    guard let requestHost = message.frameInfo.request.url?.host,
+          allowedHosts.contains(requestHost),
+          message.frameInfo.isMainFrame else {
+      log.error("Brave skus request called from disallowed host")
+      return
+    }
+    
+    guard let response = message.body as? [String: Any],
+          let methodId = response["method_id"] as? Int,
+          let data = response["data"] as? [String: Any] else {
+      log.error("Failed to retrieve method id")
+      return
+    }
+    
+    switch methodId {
+    case Method.refreshOrder.rawValue:
+      if let orderId = data["orderId"] as? String {
+        handleRefreshOrder(for: orderId, domain: requestHost, replyHandler: replyHandler)
+      }
+    case Method.fetchOrderCredentials.rawValue:
+      if let orderId = data["orderId"] as? String {
+        handleFetchOrderCredentials(for: orderId, domain: requestHost, replyHandler: replyHandler)
+      }
+    case Method.prepareCredentialsPresentation.rawValue:
+      if let domain = data["domain"] as? String, let path = data["path"] as? String {
+        handlePrepareCredentialsSummary(for: domain, path: path, replyHandler: replyHandler)
+      }
+    case Method.credentialsSummary.rawValue:
+      if let domain = data["domain"] as? String {
+        handleCredentialsSummary(for: domain, replyHandler: replyHandler)
+      }
+    default:
+      assertionFailure("Failure, the website called unhandled method with id: \(methodId)")
+    }
+  }
+  
+  private func handleRefreshOrder(for orderId: String, domain: String, replyHandler: @escaping ReplyHandler) {
+    sku?.refreshOrder(domain, orderId: orderId) { [weak self] completion in
+      do {
+        guard let data = completion.data(using: .utf8) else { return }
+        let json = try JSONSerialization.jsonObject(with: data, options: .fragmentsAllowed)
+        log.debug("skus refreshOrder")
+        self?.callback(methodId: 1, result: json, replyHandler: replyHandler)
```

Comment:
is here all the method Id is the `Method`? maybe we should use enum's rawValue for code readability

Reply:
here still didnt change? can we do
self?.callback(methodId: Method.refreshOrder.rawValue, result: json, replyHandler: replyHandler)